eventsextras.php

```diff
@@ -2,6 +2,27 @@
 
 require_once 'eventsextras.civix.php';
 use CRM_EventsExtras_ExtensionUtil as E;
+use CRM_EventsExtras_SettingsManager as SettingsManager;
+
+/**
+ * Implements hook_civicrm_buildForm().
+ *
+ * @link hhttps://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_buildForm/
+ */
+function eventsextras_civicrm_buildForm($formName, &$form){
+  if ($formName === 'CRM_Event_Form_ManageEvent_EventInfo'){
+    $eventInfoHook = new CRM_EventsExtras_Hook_BuildForm_EventInfo(SettingsManager::EVENT_INFO);
+    $eventInfoHook->buildForm($form);
+  } 
+  if ($formName === 'CRM_Event_Form_ManageEvent_Fee'){
+    $eventFeeHook = new CRM_EventsExtras_Hook_BuildForm_EventFee(SettingsManager::EVENT_FEE);
+    $eventFeeHook->buildForm($form);
+  } 
+  if ($formName === 'CRM_Event_Form_ManageEvent_Registration'){
+    $eventFeeRegistration = new CRM_EventsExtras_Hook_BuildForm_EventRegistration(SettingsManager::EVENT_REGISTRATION);
+    $eventFeeRegistration->buildForm($form);
+  }
```

Comment:
This function in time will become cluttered with IF statements and could grow much more. Can we implement something similar to here: https://github.com/compucorp/civihr/blob/master/uk.co.compucorp.civicrm.hrcore/hrcore.php#L115-L130
This way the checks for the form names and all is encapsulated within the hook class and we only have an array of listeners.